docs/usage/MTCAPIDoc.md

```diff
@@ -0,0 +1,417 @@
+
+- spec.stage: Invokes the stage operation, when set to true the migration controller switches to stage itinerary. This is a required field.
+
+- spec.verify: Specifies whether to verify the health of the migrated pods or not.
+
+
+## MigPlans
+
+```
+apiVersion: migration.openshift.io/v1alpha1
+kind: MigPlan
+metadata:
+  labels:
+    controller-tools.k8s.io: "1.0"
+  name: migplan-sample
+  namespace: openshift-migration
+spec:
+
+  srcMigClusterRef:
+    name: migcluster-local
+    namespace: openshift-migration
+
+  destMigClusterRef:
+    name: migcluster-remote
+    namespace: openshift-migration
+  
+  hooks:
+    - executionNamespace: openshift-migration
+      phase: PreBackup
+      reference:
+        name: test-6h58j
+        namespace: openshift-migration
+      serviceAccount: migration-controller
+
+  indirectImageMigration: true
+
+  indirectVolumeMigration: false
+
+  migStorageRef:
+    name: migstorage-sample
+    namespace: openshift-migration
+
+  # [!] Change namespaces to adjust which OpenShift namespaces should be migrated from source to destination cluster
+  namespaces:
+  - nginx-example
+
+  # [!] Change refresh to 'true' to force a manual reconcile
+  refresh: false
+```
+
+- spec.closed: If the migration was successful for a migplan, this value can be set True indicating that after one successful migration no new migrations can be carried out for this migplan.
+
+- spec.destMigClusterRef: ObjectReference contains enough information to let you inspect or modify the referred object.
+
+- spec.hooks: Holds a reference to a MigHook along with the desired phase to run it in.
+    -   spec.hooks.executionNamespace: Holds the name of the namespace where hooks should be implemented.
+    -   spec.hooks.phase: Indicates the phase when the hooks will be executed. Acceptable values are: PreBackup, PostBackup, PreRestore, and PostRestore.
+    -   spec.hooks.serviceAccount: Holds the name of the service account to be used for running hooks.
+    -   spec.hooks.reference: ObjectReference contains enough information to let you inspect or modify the referred object.
+
+- spec.indirectImageMigration: If set True, disables direct image migrations.
+
+- spec.indirectVolumeMigration: If set True, disables direct volume migrations.
+
+- spec.migStorageRef: ObjectReference contains enough information to let you inspect or modify the referred object.
+
+- spec.namespaces: A list of string of all the namespaces to be included in migration.
+
+- spec.persistentVolumes: Holds list of all the persistent volumes found for the namespaces included in migplan. Each entry in the list is a persistent volume with the information. Name - The PV name. Capacity - The PV storage capacity. StorageClass - The PV storage class name. Supported - Lists of what is supported. Selection - Choices made from supported. PVC - Associated PVC. NFS - NFS properties. staged - A PV has been explicitly added/updated.
```

Comment:
@eriknelson @shubham-pampattiwar 

Does the user have to populate the `persistentVolumes` section or are the fields populated by `MigStorage`?

Reply:
@apinnick It gets auto-populated,